synthesizer/program/src/parse.rs

```diff
@@ -45,58 +45,70 @@ impl<N: Network, Instruction: InstructionTrait<N>, Command: CommandTrait<N>> Par
         // Parse the semicolon ';' keyword from the string.
         let (string, _) = tag(";")(string)?;
 
+        fn intermediate<N: Network, Instruction: InstructionTrait<N>, Command: CommandTrait<N>>(
```

Comment:
Nit. Add comment describing this function.

Reply:
It would be good to explain in comments why we use this method.

Reply:
good idea; it seems that there there is some heavy-lifting done around branches right now, though, so my suggestion is to add it separately in order to avoid any merge-related issues; I've added it to my TODO, so it won't be forgotten